### PR TITLE
Add python bindings for matrix::select_k

### PR DESCRIPTION
**Pull request by benfred**

### CODE REVIEW

File: python/pylibraft/pylibraft/matrix/select_k.pyx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    >>> queries = cp.random.random_sample((n_rows, n_features),
+    ...                                   dtype=cp.float32)
+    >>> k = 40
+    >>> distances, ids = select_k(queries, k)
+    >>> distances = cp.asarray(distances)
+    >>> ids = cp.asarray(ids)
+    """
+
+    dataset_cai = cai_wrapper(dataset)
```
Comment (cjnolet) [inline python/pylibraft/pylibraft/matrix/select_k.pyx:97]: Wow- this whole function is quite elegant. For cython, anyways... but we've managed to isolate a lot of the boilerplate when working w/ mdspan/optional. It's great!

File: python/pylibraft/pylibraft/test/test_select_k.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import numpy as np
+import pytest
+
+from pylibraft.common import device_ndarray
+from pylibraft.matrix import select_k
+
+
+@pytest.mark.parametrize("n_rows", [32, 100])
+@pytest.mark.parametrize("n_cols", [40, 100])
+@pytest.mark.parametrize("k", [1, 5, 16])
```
Comment (cjnolet) [inline python/pylibraft/pylibraft/test/test_select_k.py:25]: Can we try a larger k here (some odd value like 35) or was that tripping the order of the tests?

**Review by cjnolet — APPROVED** [review]
LGTM! Really nice seeing all the hard work on the common pylibraft boilerplate paying off here. Even better seeing expansion of the Python APIs.

File: python/pylibraft/pylibraft/matrix/select_k.pyx (hunk 1 above)
Comment (benfred) [reply python/pylibraft/pylibraft/matrix/select_k.pyx:97]: I think so too!  I like how we're getting to the point where the cython code is just a thin wrapper over c++. With the CAI wrappers, the `get_dmv_float` code, the decorators to convert the output etc - I think its really getting quite nice here

**Pull request merged**